### ISSUE TITLE
`menuitemradio` role uses invalid markup

### ISSUE THREAD

**Issue by mathampson** [videojs/video.js#9052]
### Description

The markup of the Captions menu list uses `ul` with `li`s, with the role `menuitemradio` ([more info](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/menuitemradio_role)):

![Image](https://github.com/user-attachments/assets/703eb481-b496-4582-b4a9-22ad441c2931)
![Image](https://github.com/user-attachments/assets/31f631de-712a-46f3-8cf2-620329ae4268)

The same applies to the descriptions menu.

That role overrides the `li` nature of the element, causing a11y tests (e.g. Deque's Axe Core suite) to throw the counterintuitive error that the ul does not contain an li:
![Image](https://github.com/user-attachments/assets/eed5b57c-0538-48e3-8b01-92b9c489ee71)

The well-meaning attempt to use semantic markup _and_ descriptive roles therefore backfires here. We can't do both on the same element. 

Solutions include 
* nesting the two 'radio' options in a single `li` (logical enough if you regard the two links as a 'radio' set), 
* removing the `menuitemradio` role (at a slight loss of meaning to screenreaders), 
* moving the radio options outside of the `ul` (again at a slight loss of menu length signalling to screen readers)

### Reduced test case

https://leyden.elsevier.com/pages/video-player-docs

### Steps to reproduce

1. run an accessibility testing suite like Deque's Axe Core over the compiled video.js markup. Sample unit test below:

```
import { render, screen } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import VideoPlayer from './VideoPlayer';
```

```
    test('Passes accessibility tests', async () => {
        const { container } = render(<VideoPlayer options={testOptions} />);
        expect.extend(toHaveNoViolations);

        const results = await axe(container);

        expect(results).toHaveNoViolations();
    });
```

### What version of Video.js are you using?

8.16.1

### What browser(s) including version(s) does this occur with?

All

### What OS(es) and version(s) does this occur with?

All

**Comment by OwenEdwards** [comment]
This is going to need more investigation, and may even be a bug in the aXe Core; the structure of having a `<ul>` element with `role="menu"` and a child `<li>` element with `role="menuitem"` (for which `menuitemradio` is an equivalent role that _also_ conveys that it acts as one of a set of radio buttons) _is_ valid.

In fact, this HTML and ARIA is used in the W3C WAI's "[Actions Menu Button Example Using element.focus()](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-actions/)" in the ARIA Authoring Patterns Guide:

````
<ul id="menu1"
      role="menu"
      aria-labelledby="menubutton1">
    <li role="menuitem">
      Action 1
    </li>
    <li role="menuitem">
      Action 2
    </li>
    <li role="menuitem">
      Action 3
    </li>
    <li role="menuitem">
      Action 4
    </li>
  </ul>
````

It _may_ be that aXe Core doesn't like that the `menuitemradio` elements aren't contained in an element with `role="group"` (in the same way that HTML radio buttons should be contained in a `<fieldset>` with a `<legend>`), or just that it doesn't handle `menuitemradio` on an `<li>` the same way it handles `menuitem`. But whatever the cause, we should hold off "fixing" this issue in the HTML and ARIA of video.js until we confirm that it isn't a false positive from aXe Core. It would be better to note that video.js is _known_ to trigger a false positive than to always try to work around any false positive from automatic checkers.

**Comment by OwenEdwards** [comment]
Note: The following reduced code sample passes the [W3C's HTML Validator](https://validator.w3.org/nu/):

````
  <ul id="menu1"
      role="menu"
      aria-labelledby="menubutton1">
    <li role="menuitemradio" aria-checked="true">
      Option 1
    </li>
    <li role="menuitemradio" aria-checked="false">
      Option 2
    </li>
    <li role="menuitemradio" aria-checked="false">
      Option 3
    </li>
    <li role="menuitemradio" aria-checked="false">
      Option 4
    </li>
  </ul>
````

**Comment by mathampson** [comment]
> may even be a bug in the aXe Core

@OwenEdwards Thanks for looking into this, and you make a really good point.

A key thing to note is `menuitem` is accepted, due to the explicit `menu` role on the `ul`, but `menuitemradio` is not.  By my understanding you _are_ allowed to mix those roles in a `menu`, so why does one throw and not the other?

Here's a test case including a `menuitem` which also passes the validator FWIW:

```
<ul id="menu1"
      role="menu">
    <li role="menuitem">
      Option 1
    </li>
    <li role="menuitemradio" aria-checked="false">
      Option 2
    </li>
    <li role="menuitemradio" aria-checked="false">
      Option 3
    </li>
    <li role="menuitemradio" aria-checked="false">
      Option 4
    </li>
  </ul>
```

I've looked for any issue or SO question regarding aXe Core again, and not found this exact example. [There's some relevent discussion here](https://github.com/dequelabs/axe-core/issues/1942), but that implies as long as the roles are valid among themselves (as they are in video.js) and ditto the HTML, all should be ok.

Would you like me to [raise this with aXe](https://github.com/dequelabs/axe-core/issues) and see what they say?

**Comment by adubouski** [comment]
@OwenEdwards the thing is the UL in video.js does not have "menu" role, that is why AXE complains, as far as I understand:

<img width="540" height="113" alt="Image" src="https://github.com/user-attachments/assets/206aa855-cb85-4531-93d9-f016d54f8cae" />

Seems like "menu" role is set for captions, but not for descriptions.
